src/Integration/Notifications/SonarQubeNotifications.cs

```diff
@@ -0,0 +1,145 @@
+    {
+        private readonly ITimer timer;
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly CancellationTokenSource cancellation = new CancellationTokenSource();
+
+        private DateTimeOffset lastCheckDate;
+        private string projectKey;
+
+        public INotificationIndicatorViewModel Model { get; private set; }
+
+        public NotificationData GetNotificationData() =>
+            new NotificationData
+                {
+                    IsEnabled = Model.AreNotificationsEnabled,
+                    LastNotificationDate = lastCheckDate
+                };
+
+        [ImportingConstructor]
+        [ExcludeFromCodeCoverage] // Do not unit test MEF constructor
+        internal SonarQubeNotifications(IHost host)
+            : this(host.SonarQubeService, new NotificationIndicatorViewModel(),
+                  new TimerWrapper { Interval = 60000 })
+        {
+        }
+
+        internal SonarQubeNotifications(ISonarQubeService sonarQubeService,
+            INotificationIndicatorViewModel model, ITimer timer)
+        {
+            this.timer = timer;
+            this.sonarQubeService = sonarQubeService;
+            Model = model;
+
+            timer.Elapsed += OnTimerElapsed;
+        }
+
+        public async Task StartAsync(string projectKey, NotificationData notificationData)
+        {
+            if (projectKey == null)
+            {
+                throw new ArgumentNullException(nameof(projectKey));
+            }
+
+            this.projectKey = projectKey;
+
+            Model.AreNotificationsEnabled = notificationData?.IsEnabled ?? true;
+
+            var oneDayAgo = DateTimeOffset.Now.AddDays(-1);
+            if (notificationData == null ||
+                notificationData.LastNotificationDate < oneDayAgo)
+            {
+                lastCheckDate = oneDayAgo;
+            }
+            else
+            {
+                lastCheckDate = notificationData.LastNotificationDate;
+            }
+
+            await UpdateEvents();
+            timer.Start();
+        }
+
+        public void Stop()
+        {
+            cancellation.Cancel();
+
+            timer.Stop();
+            Model.IsIconVisible = false;
+        }
+
+        private async Task UpdateEvents()
+        {
+            if (!sonarQubeService.IsConnected)
+            {
+                return;
+            }
+
+            var events = await GetNotificationEvents();
+            if (events == null)
+            {
+                Stop();
+                return;
+            }
+            Model.IsIconVisible = true;
+            Model.SetNotificationEvents(events);
+        }
+
+        private async void OnTimerElapsed(object sender, EventArgs e)
+        {
+            await UpdateEvents();
+        }
+
+        private async Task<IList<SonarQubeNotification>> GetNotificationEvents()
+        {
+            var events = await sonarQubeService.GetNotificationEventsAsync(projectKey,
```

Comment:
Just as a reminder, methods on sonarQubeService can throw exceptions (timeout, cancellation, other errors) and you might want to do something in case this happens.